src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -40,13 +40,36 @@ void ComponentFiller::addVariables(Solver::Modeler::Api::ILinearProblem& pb,
                                    Solver::Modeler::Api::LinearProblemData& data,
                                    Solver::Modeler::Api::FillContext& ctx)
 {
-    auto evaluator = std::make_unique<Solver::Visitors::EvalVisitor>(evaluationContext_);
+    Solver::Visitors::EvalVisitor evaluator(evaluationContext_);
     for (const auto& variable: component_.getModel()->Variables() | std::views::values)
     {
-        pb.addVariable(evaluator->dispatch(variable.LowerBound().RootNode()),
-                       evaluator->dispatch(variable.UpperBound().RootNode()),
-                       variable.Type() != Study::SystemModel::ValueType::FLOAT,
-                       component_.Id() + "." + variable.Id());
+        const auto variableID = component_.Id() + "." + variable.Id();
+        switch (variable.Type())
+        {
+        case Study::SystemModel::ValueType::BOOL:
+        {
+            pb.addVariable(0., 1., true, variableID);
```

Comment:
the user might define lb / ub (for example ub can be a model parameter). it's only in the case where the user doesn't define lb/ub that we should us these default values. maybe set them in the Variable object model to keep this code simple?